workflows/pipe-common/scripts/mount_storage.py

```diff
@@ -592,7 +596,8 @@ def build_mount_command(self, params):
         return command
 
     def append_timeout_options(self, mount_options):
-        if self.share_mount.mount_type == 'SMB':
+        if self.share_mount.mount_type == 'SMB' or \
+                (PermissionHelper.is_run_sensitive() and NFSMounter.sensitive_policy != "TIMEOUT"):
```

Comment:
Condition should be `mount_type == SMB  or not PermissionHelper.is_run_sensitive() or NFSMounter.sensitive_policy != "TIMEOUT"` (do not apply timeout for non-sensitive runs)